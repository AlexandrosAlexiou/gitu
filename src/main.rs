use std::{
    error::Error,
    io::{stderr, BufWriter},
};

use clap::Parser;
use crossterm::{
    terminal::{disable_raw_mode, enable_raw_mode, EnterAlternateScreen, LeaveAlternateScreen},
    ExecutableCommand,
};
use ratatui::{prelude::CrosstermBackend, Terminal};

pub fn main() -> Result<(), Box<dyn Error>> {
    let mut terminal = Terminal::new(CrosstermBackend::new(BufWriter::new(stderr())))?;
    terminal.hide_cursor()?;
    enable_raw_mode()?;
    stderr().execute(EnterAlternateScreen)?;

    let result = gitu::run(gitu::cli::Args::parse(), &mut terminal);

    stderr().execute(LeaveAlternateScreen)?;
    disable_raw_mode()?;

    result?;

    Ok(())
}
