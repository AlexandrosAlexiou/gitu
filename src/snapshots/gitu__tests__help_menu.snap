---
source: src/main.rs
expression: terminal.backend().buffer()
---
Buffer {
    area: Rect { x: 0, y: 0, width: 70, height: 12 },
    content: [
        "                                                                      ",
        "──────────────────────────────────────────────────────────────────────",
        "Help                 Submenu                                          ",
        "g Refresh            h Help                                           ",
        "tab ToggleSection    b Branch                                         ",
        "k SelectPrevious     c Commit                                         ",
        "j SelectNext         f Fetch                                          ",
        "C-u HalfPageUp       l Log                                            ",
        "C-d HalfPageDown     F Pull                                           ",
        "y ShowRefs           P Push                                           ",
        "                     r Rebase                                         ",
        "                     K Discard                                        ",
    ],
    styles: [
        x: 0, y: 0, fg: Reset, bg: Rgb(80, 73, 69), underline: Reset, modifier: NONE,
        x: 0, y: 1, fg: Rgb(80, 73, 69), bg: Reset, underline: Reset, modifier: NONE,
        x: 0, y: 2, fg: Rgb(125, 174, 163), bg: Reset, underline: Reset, modifier: BOLD,
        x: 4, y: 2, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 21, y: 2, fg: Rgb(125, 174, 163), bg: Reset, underline: Reset, modifier: BOLD,
        x: 28, y: 2, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 0, y: 3, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 1, y: 3, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 21, y: 3, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 22, y: 3, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 0, y: 4, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 3, y: 4, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 21, y: 4, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 22, y: 4, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 0, y: 5, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 1, y: 5, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 21, y: 5, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 22, y: 5, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 0, y: 6, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 1, y: 6, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 21, y: 6, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 22, y: 6, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 0, y: 7, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 3, y: 7, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 21, y: 7, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 22, y: 7, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 0, y: 8, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 3, y: 8, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 21, y: 8, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 22, y: 8, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 0, y: 9, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 1, y: 9, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 21, y: 9, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 22, y: 9, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 21, y: 10, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 22, y: 10, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 21, y: 11, fg: Rgb(211, 134, 155), bg: Reset, underline: Reset, modifier: NONE,
        x: 22, y: 11, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
    ]
}
